lib/core-classes/DecodersService.ts

```diff
@@ -1,31 +1,64 @@
-import { PluginContext, EmbeddedSDK, Plugin } from "kuzzle";
+import { PayloadService } from './PayloadService';
+import { PluginContext, EmbeddedSDK, Plugin, PluginApiDefinition } from "kuzzle";
 
 import { Decoder } from "./Decoder";
 import { DeviceManagerConfig } from "../DeviceManagerPlugin";
 import { DecoderContent } from '../types/decoders/DecodersContent';
+import { PayloadHandler } from 'index';
 
 export class DecodersService {
   private config: DeviceManagerConfig;
   private context: PluginContext;
 
-  private decoders: Map<string, Decoder>;
+  private _decoders: Map<string, Decoder>;
+  private handlers: { decoder: Decoder, handler: PayloadHandler }[];
 
   get sdk(): EmbeddedSDK {
     return this.context.accessors.sdk;
   }
 
-  constructor(plugin: Plugin, decoders: Map<string, Decoder>) {
+  get decoders(): Map<string, Decoder> {
+    return this._decoders;
+  }
+
+  constructor(plugin: Plugin, decoders: { decoder: Decoder, handler: PayloadHandler }[]) {
     this.config = plugin.config as any;
     this.context = plugin.context;
+    this.handlers = decoders
+    this._decoders = new Map();
+    
+    
+    for (const { decoder } of this.handlers) {
+      this.register(decoder);
+    }
 
-    this.decoders = decoders;
+    this.printDecoders();
   }
 
   async list(): Promise<DecoderContent[]> {
     const decoders = Array
-      .from(this.decoders.values())
+      .from(this._decoders.values())
       .map(decoder => decoder.serialize())
 
     return decoders;
   }
+
+  printDecoders() {
+    for (const decoder of this._decoders.values()) {
+      this.context.log.info(`Register API action "device-manager/payload:${decoder.action}" with decoder "${decoder.constructor.name}" for device "${decoder.deviceModel}"`);
+    }
+  }
+
+  register(decoder: Decoder) {
+    this.decoders.set(decoder.deviceModel, decoder);
+  }
+
+  registerPayloadController(api: PluginApiDefinition, payloadService: PayloadService) {
```

Comment:
It's better in term of architecture but still I'm not completely satisfied with this solution for 2 reasons:
  - this class have to modify another class internal state (`api` object)
  - the controller definition can not be used somewhere else (for example in an custom PayloadController with other actions)